hera_qm/tests/test_xrfi.py

```diff
@@ -1174,6 +1183,101 @@ def test_xrfi_run(tmpdir):
     for fname in [ocal_file, acal_file, model_file, raw_dfile]:
         os.remove(fname)
 
+def test_xrfi_run_multifile(tmpdir):
+    # test xrfi_run with multiple files
+    # The warnings are because we use UVFlag.to_waterfall() on the total chisquareds
+    # This doesn't hurt anything, and lets us streamline the pipe
+    mess1 = ['This object is already a waterfall']
+    messages = 8 * mess1
+    cat1 = [UserWarning]
+    categories = 8 * cat1
+    # Spoof a couple files to use as extra inputs (xrfi_run needs two cal files and two data-like files)
+    tmp_path = tmpdir.strpath
+    fake_obses = ['zen.2457698.40355191.HH', 'zen.2457698.40367619.HH', 'zen.2457698.40380046.HH']
+    ocal_files = []
+    acal_files = []
+    model_files =[]
+    raw_dfiles = []
+    for uvf, cf, fo in zip(test_uvh5_files, test_c_files, fake_obses):
+        ocal_file = os.path.join(tmp_path, fo + '.omni.calfits')
+        shutil.copyfile(cf, ocal_file)
+        ocal_files.append(ocal_file)
+        acal_file = os.path.join(tmp_path, fo + '.abs.calfits')
+        shutil.copyfile(cf, acal_file)
+        acal_files.append(acal_file)
+        raw_dfile = os.path.join(tmp_path, fo + '.uvh5')
+        shutil.copyfile(uvf, raw_dfile)
+        raw_dfiles.append(raw_dfile)
+        model_file = os.path.join(tmp_path, fo + '.omni_vis.uvh5')
+        shutil.copyfile(uvf, model_file)
+        model_files.append(model_file)
+
+    # check warnings
+    with pytest.warns(None) as record:
+        xrfi.xrfi_run(ocal_files, acal_files, model_files, raw_dfiles,
+                      'Just a test', kt_size=3, cross_median_filter=True)
+    assert len(record) >= len(messages)
+    n_matched_warnings = 0
+    for i in range(len(record)):
+        if mess1[0] in str(record[i].message) and cat1[0] == record[i].category:
+            n_matched_warnings += 1
+    assert n_matched_warnings == 8
+    outdirs = ['zen.2457698.40355191.xrfi',
```

Comment:
If you add back in the feature for not flagging edge files, then I think we should run this twice, once with it on and once with it off. In the first case, this should produce one output file (depending on Kt) and in the later case, this should produce 3 files.

Reply:
Good idea. I'll add this test.

Reply:
i end up checking whether all times in flag zone are flagged since there aren't enough integrations to test whether files haven't been written.